src/main/java/io/reactivex/Observable.java

```diff
@@ -420,4 +422,129 @@ public static int bufferSize() {
             return v;
         });
     }
+    
+    public final Observable<T> observeOn(Scheduler scheduler) {
+        return observeOn(scheduler, false, bufferSize());
+    }
+    
+    public final Observable<T> observeOn(Scheduler scheduler, boolean delayError) {
+        return observeOn(scheduler, delayError, bufferSize());
+    }
+    
+    public final Observable<T> observeOn(Scheduler scheduler, boolean delayError, int bufferSize) {
+        // TODO implement
+        throw new UnsupportedOperationException();
+    }
+    
+    public final Observable<T> subscribeOn(Scheduler scheduler) {
+        // TODO implement
+        throw new UnsupportedOperationException();
+    }
+    
+    public final Observable<T> unsubscribeOn(Scheduler scheduler) {
+        // TODO implement
+        throw new UnsupportedOperationException();
+    }
+    
+    public final Observable<T> cache() {
+        return CachedObservable.from(this);
+    }
+
+    public final Observable<T> cache(int capacityHint) {
+        return CachedObservable.from(this, capacityHint);
+    }
+
+    public final ConnectableObservable<T> publish() {
+        return publish(bufferSize());
+    }
+
+    public final ConnectableObservable<T> publish(int bufferSize) {
+        if (bufferSize <= 0) {
+            throw new IllegalArgumentException("bufferSize > 0 required but it was " + bufferSize);
+        }
+        return OperatorPublish.create(this, bufferSize);
+    }
+
+    public final <R> Observable<R> publish(Function<? super Observable<T>, ? extends Observable<R>> selector) {
```

Comment:
As per #1353 I recommend something other than "selector" for the parameter name (here and elsewhere) "since these functions aren't really 'selecting' items so much as changing them."